mindmeld/active_learning/alp.py

```diff
@@ -0,0 +1,234 @@
+        training_strategies: list,
+        training_level: str,
+        log_selection_strategy: str,
+        save_sampled_queries: bool,
+        early_stopping_window: int,
+        log_usage_pct: float,
+        labeled_logs_pattern: str,
+        unlabeled_logs_path: str,
+        output_folder: str,
+    ):
+        """
+        Args:
+            app_path (str): Path to MindMeld application
+            train_pattern (str): Regex pattern to match train files. For example, ".*train.*.txt"
+            test_pattern (str): Regex pattern to match test files. For example, ".*test.*.txt"
+            load (bool): Whether to load pickled queries from a local folder
+            save (bool): Whether to save queries as a local pickle file
+            train_seed_pct (float): Percentage of training data to use as the initial seed
+            n_classifiers (int): Number of classifiers to be used by a subset of heuristics
+            n_epochs (int): Number of epochs to run training
+            batch_size (int): Number of queries to select at each iteration
+            training_level (str): The hierarchy level to train ("domain" or "intent")
+            training_strategies (List[str]): List of strategies to use for training
+            log_selection_strategy (str): Single strategy to use for log selection
+            save_sampled_queries (bool): Whether to save the queries sampled at each iteration
+            early_stopping_window (int): If the drops for n iterations, terminate training early
+            log_usage_pct (float): Percentage of the log data to use for selection
+            labeled_logs_pattern (str): Pattern to obtain logs already labeled in a MindMeld app
+            unlabeled_logs_path (str): Path a logs text file with unlabeled queries
+            output_folder (str):
+        """
+        self.app_path = app_path
+        self.train_pattern = train_pattern
+        self.test_pattern = test_pattern
+        self.load = load
+        self.save = save
+        self.train_seed_pct = train_seed_pct
+        self.n_classifiers = n_classifiers
+        self.n_epochs = n_epochs
+        self.batch_size = batch_size
+        self.training_level = training_level
+        self.training_strategies = training_strategies
+        self.log_selection_strategy = log_selection_strategy
+        self.save_sampled_queries = save_sampled_queries
+        self.early_stopping_window = early_stopping_window
+        self.log_usage_pct = log_usage_pct
+        self.labeled_logs_pattern = labeled_logs_pattern
+        self.unlabeled_logs_path = unlabeled_logs_path
+
+        self.output_manager = self.get_output_manager(output_folder)
+        self.mindmeld_classifier = self.get_classifier()
+        self.init_data_bucket = None
+        self.data_bucket = None
+
+    def get_output_manager(self, output_folder):
+        return OutputManager(
+            active_learning_params=deepcopy(self.__dict__), output_folder=output_folder
```

Comment:
might be a better idea to overload the to_dict method of this class and then pass in the relevant info. A lot of redundant into will be passed to the `OutputManager ` if we just use `__dict__`

Reply:
Made a new function `as_dict()`instead of overloading `__dict__()` after running into the following error: "`__dict__()` is not callable".